File: Templates/Day_1/HelloWorld/SampleSolutions/hello_mpi.c
#include <stdio.h>
#include <mpi.h>


int main(int argc, char **argv)
{
  MPI_Init(&argc, &argv); // alt.: NULL,NULL
  
  int size, rank;

  // Copy the communicator
  MPI_Comm my_world = MPI_COMM_WORLD;

  MPI_Comm_size(my_world, &size);
  MPI_Comm_rank(my_world, &rank);

  printf("I am rank %d out of %d!\n", rank, size);

  MPI_Finalize();

  return 0;
}

File: Templates/Day_1/HelloWorld/SampleSolutions/hello_mpi.f90
program Mpihello

  use mpi

  Implicit none

  Integer :: world_size, world_rank
  Integer :: my_world, ierr

  Call MPI_Init(ierr)

  my_world = MPI_COMM_WORLD

  Call MPI_Comm_size(my_world, world_size, ierr)
  Call MPI_Comm_rank(my_world, world_rank, ierr)

  print *,"I am rank ", world_rank," out of", world_size
 
  Call MPI_Finalize(ierr)
end program Mpihello

